### DISCUSSION TITLE
Why bump minimum Go version if not using newer features?

### DISCUSSION THREAD

**Discussion by mfridman** [ClickHouse/clickhouse-go#1397] [General]
I noticed in https://github.com/ClickHouse/clickhouse-go/pull/1378 the minimum Go version got bumped to `go1.22`, but afaics no new language features are (currently) being used.

Given this module may be used in other projects, this forces a cascading upgrade on said projects that aim to support slightly older versions of Go, e.g., `go1.21`.

I appreciate the Go team aims to support the last two major releases, but libraries in the community have the affordance to keep backwards compatible for maximum compatibility.

**Comment by jkaflik** [comment]
Hi @mfridman 

You are absolutely right. I will update minimum required version in https://github.com/ClickHouse/clickhouse-go/pull/1405

**Reply by mfridman** [reply]
Thanks @jkaflik, I appreciate it. 

Whenever ya'll start using newer language features that need a version bump I totally understand ✌️.
